### PR TITLE
feature/ DAO-285 

### PR DESCRIPTION
**Pull request by sepehr2github**
- Add Deposit
- Add Withdraw
- Add transactions
- Add FinanceSideCard
- Refactor action details page and New transaction page
- Add Custom Content Switcher

### CODE REVIEW

File: packages/govern-console/src/components/ActionBuilder/Screens/AssetWithdrawal.tsx [hunk 1]
```diff
@@ -80,6 +80,7 @@ export const AssetWithdrawal: React.FC<AssetWithdrawalProps> = ({ onClick }) =>
 
       const executor = dao?.executor.address;
       const values = [asset.address, executor, recipient, asset.amount, reference];
+      console.log('seeValues', withdrawalAmount, withdrawalAssets[token], asset);
```
Comment (githubdoramon) [inline packages/govern-console/src/components/ActionBuilder/Screens/AssetWithdrawal.tsx]: Pls remove the console log

File: packages/govern-console/src/containers/DAO/DaoHomePage.tsx [hunk 2]
```diff
@@ -128,25 +128,29 @@ const DaoHomePage: React.FC = () => {
...
-            {/* TODO: Operation not found page */}
-            <ApmRoute render={() => <div>Operation not found on dao. Go home?</div>} />
+            <ApmRoute exact path={`${path}actions/executions/:id`} component={ProposalDetails} />
+            <ApmRoute exact path={`${path}actions/new`} component={NewExecution} />
+            <ApmRoute render={() => history.push('/not-found')} />
           </Switch>
         </Suspense>
       </StyledGridItem>
-      {pathname === `${url}/settings` && (
+      {(pathname === `${url}/settings` || pathname === `${url}/actions/new`) && (
```
Comment (githubdoramon) [inline packages/govern-console/src/containers/DAO/DaoHomePage.tsx]: Let's extract the path names to constants, so it is easier to do this type of comparison

File: packages/govern-console/src/containers/DAO/components/DaoTransactionCard/DaoTransactionCard.tsx [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+          </Time>
+        </TextContainer>
+      </InfoContainer>
+      <PriceContainer className={info?.typename.toLowerCase()}>{Price}</PriceContainer>
+    </ActionCard>
+  );
+};
+
+export default DaoTransactionCard;
+// info?.typename ? (
```
Comment (githubdoramon) [inline packages/govern-console/src/containers/DAO/components/DaoTransactionCard/DaoTransactionCard.tsx]: Remove unused comments

File: packages/govern-console/src/containers/DAO/components/ModalContents/SignTransfer.tsx [hunk 4]
```diff
@@ -0,0 +1,348 @@
...
+const SignDeposit: React.FC = () => {
+  const history = useHistory();
+  const { daoIdentifier } = useTransferContext();
+
+  const { getValues } = useFormContext();
+  const { networkName } = useWallet();
+  const { transactions } = useTransferContext();
+  const [txState, setTxState] = useState(SigningState.Processing);
+  const [transactionList, setTransactionList] = useState<CustomTransaction[]>([...transactions]);
+  // const [errorMessage, setErrorMessage] = useState('');
```
Comment (githubdoramon) [inline packages/govern-console/src/containers/DAO/components/ModalContents/SignTransfer.tsx]: Remove line

File: packages/govern-console/src/containers/DAO/components/ModalContents/SignTransfer.tsx [hunk 5]
```diff
@@ -0,0 +1,348 @@
...
+      // TODO: implement a solution without full page reload
+      window.location.reload();
+    }
+  }, [containerHash, daoIdentifier, history]);
+
+  const SendToExplore = useCallback(() => {
+    window.open(
+      `${
+        {
+          Rinkeby: 'https://rinkeby.etherscan.io/tx/',
```
Comment (githubdoramon) [inline packages/govern-console/src/containers/DAO/components/ModalContents/SignTransfer.tsx]: Can't this come from use-wallet?

File: packages/govern-console/src/containers/DAO/components/ModalContents/SignTransfer.tsx [hunk 6]
```diff
@@ -0,0 +1,348 @@
...
+        updateStatus(index, CustomTransactionStatus.Successful);
+        setContainerHash(Proposal.getContainerHashFromReceipt(txReceipt, ReceiptType.Scheduled));
+        setTransactionHash(txReceipt.transactionHash);
+      } catch (ex: any) {
+        isQueueAborted = true;
+        updateStatus(index, CustomTransactionStatus.Failed);
+        setTxState(SigningState.Failure);
+
+        console.log(ex);
+        // setErrorMessage(ex.message);
```
Comment (githubdoramon) [inline packages/govern-console/src/containers/DAO/components/ModalContents/SignTransfer.tsx]: remove console.log and re-add error

File: packages/govern-console/src/containers/DAO/components/ModalContents/SignTransfer.tsx (hunk 4 above)
Comment (githubdoramon) [reply packages/govern-console/src/containers/DAO/components/ModalContents/SignTransfer.tsx]: if not used for course.... but if the commented line below comes back, this is needed

File: packages/govern-console/src/containers/DAO/components/ModalContents/components/SelectToken/TokenCard.tsx [hunk 7]
```diff
@@ -0,0 +1,77 @@
+import styled from 'styled-components';
+
+import ETHIcon from 'images/pngs/eth_logo.png';
+import { ASSET_ICON_BASE_URL } from 'utils/constants';
+import { getTruncatedAccountAddress } from 'utils/account';
+
+const logoAddresses: any = {
```
Comment (githubdoramon) [inline packages/govern-console/src/containers/DAO/components/ModalContents/components/SelectToken/TokenCard.tsx]: logoAdresses looks a confusing name... should be tokenAddresses right

File: packages/govern-console/src/containers/NewExecution/NewExecution.tsx [hunk 8]
```diff
@@ -159,81 +163,81 @@ const NewExecution: React.FC = () => {
   };
 
   return (
-    <PageContent pageName={PageName.NEW_EXECUTION}>
-      <Box>
-        <BackButton onClick={() => history.goBack()}>
-          <img src={backButtonIcon} />
-        </BackButton>
-        <Grid>
+    // <PageContent pageName={PageName.NEW_EXECUTION}>
```
Comment (githubdoramon) [inline packages/govern-console/src/containers/NewExecution/NewExecution.tsx]: remove comment

File: packages/govern-console/src/services/finances.ts [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+    return {
+      price: null,
+      error: { code: jsonResponse.code, message: jsonResponse.message },
+    };
+  } catch (err) {
+    console.log('ERROR-Fetching price', err);
+  }
+}
+
+/**
```
Comment (githubdoramon) [inline packages/govern-console/src/services/finances.ts:29]: Pls add comment here that this is related to KPI options migration, and only 2 DAOs did it. link to https://blog.aragon.org/uma-kpi-options-airdrop-now-live-for-aragon-govern-daos/

File: packages/govern-console/src/utils/ipfs.tsx [hunk 10]
```diff
@@ -131,57 +131,62 @@ export async function fetchIPFS(uriOrCid: string) {
...
+              try {
+                data.text = new TextDecoder().decode(buffer);
+              } catch (err) {}
+            }
+          }
         }
       }
     }
+  } catch {
+    console.log('Fetch failed');
```
Comment (githubdoramon) [inline packages/govern-console/src/utils/ipfs.tsx]: maybe console.error is better?

File: packages/govern-console/src/utils/urls.tsx [hunk 11]
```diff
@@ -16,5 +16,5 @@ export function settingsUrl(daoName: string): string {
  * @returns {string} the url of the proposal details page
  */
 export function proposalDetailsUrl(daoName: string, containerHash: string): string {
-  return `/daos/${daoName}/executions/${containerHash}`;
+  return `/daos/${daoName}/actions/executions/${containerHash}`;
```
Comment (githubdoramon) [inline packages/govern-console/src/utils/urls.tsx:19]: We could have all URL constants in this file maybe?

**Review by githubdoramon — CHANGES_REQUESTED** [review]

File: packages/govern-console/src/utils/urls.tsx [hunk 12]
```diff
@@ -26,3 +26,10 @@ export function proposalSettingsUrl(url: string): string {
 export function proposalNewActionsUrl(url: string): string {
   return `${url}/actions/new`;
 }
+
+export const DoaNotFoundUrl = `/daos/not-found`;
```
Comment (githubdoramon) [inline packages/govern-console/src/utils/urls.tsx]: Tiny typo "DOA" Vs "DAO"

Comment (sepehr2github) [reply packages/govern-console/src/utils/urls.tsx]: @githubdoramon Typo Fixed!

**Review by githubdoramon — APPROVED** [review]

**Pull request merged**